src/LinuxTracing/ThreadStateManager.h

```diff
@@ -60,10 +62,20 @@ class ThreadStateManager {
 
  private:
   struct OpenState {
-    OpenState(orbit_grpc_protos::ThreadStateSlice::ThreadState state, uint64_t begin_timestamp_ns)
-        : state{state}, begin_timestamp_ns{begin_timestamp_ns} {}
+    OpenState(orbit_grpc_protos::ThreadStateSlice::ThreadState state, uint64_t begin_timestamp_ns,
+              pid_t was_blocked_by_thread, pid_t was_blocked_by_process)
+        : state{state},
+          begin_timestamp_ns{begin_timestamp_ns},
+          was_blocked_by_thread{was_blocked_by_thread},
+          was_blocked_by_process{was_blocked_by_process} {}
     orbit_grpc_protos::ThreadStateSlice::ThreadState state;
     uint64_t begin_timestamp_ns;
+    // The next two fields are optional, we use them to transfer info between interruptible sleep
+    // state (gray) and runnable state (blue) so we can know which tid and pid caused the thread to
+    // turn into a runnable state if you don't need them just put zeros when using the OpenState
```

Comment:
Erase "if you don't need them .... ". Actually I wouldn't do that and rather set the default values in the constructor. In that sense, you don't need to add "0, 0" in all the cases we don't use them.

Reply:
Done.